src/index.ts

```diff
@@ -0,0 +1,2721 @@
+  var c = a.add(b).toNumber();
+  if (Number.isFinite(c) && !Decimal.eq_tolerance(c, a.toNumber()+b.toNumber()))
+  {
+      console.log(a + ", " + b);
+  }
+}
+
+for (var i = 0; i < 100; ++i)
+{
+  var a = Decimal.randomDecimalForTesting(Math.round(Math.random()*4));
+  var b = Decimal.randomDecimalForTesting(Math.round(Math.random()*4));
+  if (Math.random() > 0.5) { a = a.recip(); }
+  if (Math.random() > 0.5) { b = b.recip(); }
+  var c = a.mul(b).toNumber();
+  if (Number.isFinite(c) && Number.isFinite(a.toNumber()) && Number.isFinite(b.toNumber()) && a.toNumber() != 0 && b.toNumber() != 0 && c != 0 && !Decimal.eq_tolerance(c, a.toNumber()*b.toNumber()))
+  {
+      console.log("Test 1: " + a + ", " + b);
+  }
+  else if (!Decimal.mul(a.recip(), b.recip()).eq_tolerance(Decimal.mul(a, b).recip()))
+  {
+      console.log("Test 3: " + a + ", " + b);
+  }
+}
+
+for (var i = 0; i < 10; ++i)
+{
+  var a = Decimal.randomDecimalForTesting(Math.round(Math.random()*4));
+  var b = Decimal.randomDecimalForTesting(Math.round(Math.random()*4));
+  if (Math.random() > 0.5 && a.sign !== 0) { a = a.recip(); }
+  if (Math.random() > 0.5 && b.sign !== 0) { b = b.recip(); }
+  var c = a.pow(b);
+  var d = a.root(b.recip());
+  var e = a.pow(b.recip());
+  var f = a.root(b);
+
+  if (!c.eq_tolerance(d) && a.sign !== 0 && b.sign !== 0)
+  {
+    console.log("Test 1: " + a + ", " + b);
+  }
+  if (!e.eq_tolerance(f) && a.sign !== 0 && b.sign !== 0)
+  {
+    console.log("Test 2: " + a + ", " + b);
+  }
+}
+
+for (var i = 0; i < 10; ++i)
+{
+  var a = Math.round(Math.random()*18-9);
+  var b = Math.round(Math.random()*100-50);
+  var c = Math.round(Math.random()*18-9);
+  var d = Math.round(Math.random()*100-50);
+  console.log("Decimal.pow(Decimal.fromMantissaExponent(" + a + ", " + b + "), Decimal.fromMantissaExponent(" + c + ", " + d + ")).toString()");
+}
+
+*/
+
+  //Pentation/pentate: The result of tetrating 'height' times in a row. An absurdly strong operator - Decimal.pentate(2, 4.28) and Decimal.pentate(10, 2.37) are already too huge for break_eternity.js!
+  // https://en.wikipedia.org/wiki/Pentation
+  public pentate(height = 2, payload: DecimalSource = FC_NN(1, 0, 1)): Decimal {
+    payload = D(payload);
+    const oldheight = height;
+    height = Math.trunc(height);
+    const fracheight = oldheight - height;
+
+    //I have no idea if this is a meaningful approximation for pentation to continuous heights, but it is monotonic and continuous.
+    if (fracheight !== 0) {
+      if (payload.eq(Decimal.dOne)) {
+        ++height;
+        payload = new Decimal(fracheight);
+      } else {
+        if (this.eq(10)) {
+          payload = payload.layeradd10(fracheight);
+        } else {
+          payload = payload.layeradd(fracheight, this);
+        }
+      }
+    }
+
+    for (let i = 0; i < height; ++i) {
+      payload = this.tetrate(payload); // FIXME: This seems to be a bug - tetrate only handles numbers but this is a decimal
```

Comment:
Not sure what to do about this one. As far as I can tell, `tetrate` expects a number for its first parameter, and it does not do any Decimal conversion. `payload` is a Decimal here, though.

This looks like a bug. `tetrate` checks `<`, equality to a number constant, and does `Math.trunc` on `height`, so it seems like this code won't work as-is.

I believe it would work converting payload to a number,  

```suggestion
      payload = this.tetrate(payload.toNumber());
```

but I don't know the implications of that. Maybe `tetrate` needs to take a `DecimalSource` for the height, and do conversion before the checks?